qa/tasks/vstart_runner.py

```diff
@@ -361,12 +361,25 @@ def sudo_write_file(self, path, data, **kwargs):
         """
         self.write_file(path, data, sudo=True, **kwargs)
 
-    def _perform_checks_and_return_list_of_args(self, args, omit_sudo):
-        # Since Python's shell simulation can only work when commands are
-        # provided as a list of argumensts...
-        if isinstance(args, str):
-            args = args.split()
+    def _perform_checks_when_str(self, args, omit_sudo, shell):
+        # We'll let sudo be a part of command even omit flag says otherwise in
+        # cases of commands which can normally be ran only by root.
+        if 'sudo' in args:
+            for x in ('-u', 'passwd', 'chown'):
+                if args.find(x) != -1:
+                    omit_sudo = False
+
+            if omit_sudo:
+                args.replace('sudo', '')
 
+        args = args.replace('adjust-ulimits','')
+        args = args.replace('ceph-coverage', '')
```

Comment:
Let's remove these and just override the commands in the shell script we run (comment later).

Reply:
Replied on second thread of this conversation, leave that one open and closing this one.